server/src/main/java/org/opensearch/autoforcemerge/AutoForceMergeManager.java

```diff
@@ -0,0 +1,427 @@
+                isRemoteStoreEnabled = isRemoteStorageEnabled();
+                initialCheckDone.set(true);
+            }
+        }
+
+        /**
+         * Checks if remote storage is enabled in the cluster settings.
+         */
+        private boolean isRemoteStorageEnabled() {
+            return clusterService.getSettings().getAsBoolean(REMOTE_CLUSTER_STATE_ENABLED_SETTING.getKey(), false);
+        }
+
+        /**
+         * Checks if cluster has warm nodes.
+         */
+        private boolean hasWarmNodes() {
+            if (hasWarmNodes == true) return true;
+            ClusterState clusterState = clusterService.state();
+            return hasWarmNodes = clusterState.getNodes().getNodes()
+                .values()
+                .stream()
+                .anyMatch(DiscoveryNode::isWarmNode
+                );
+        }
+    }
+
+    /**
+     * Validates node-level conditions for force merge operations.
+     * This validator checks CPU usage, JVM memory usage, and force merge thread availability
+     * to determine if force merge operations can proceed safely.
+     */
+    protected class NodeValidator implements ValidationStrategy {
+
+        private int maxConcurrentForceMerges;
+
+        @Override
+        public ValidationResult validate() {
+            double cpuPercent = osService.stats().getCpu().getPercent();
+            if (cpuPercent >= forceMergeManagerSettings.getCpuThreshold()) {
+                logger.debug("CPU usage too high: {}%", cpuPercent);
+                return new ValidationResult(false);
+            }
+            double jvmUsedPercent = jvmService.stats().getMem().getHeapUsedPercent();
+            if (jvmUsedPercent >= forceMergeManagerSettings.getJvmThreshold()) {
+                logger.debug("JVM memory usage too high: {}%", jvmUsedPercent);
+                return new ValidationResult(false);
+            }
+            if (areForceMergeThreadsAvailable() == false) {
+                logger.debug("No force merge threads available");
+                return new ValidationResult(false);
+            }
+            return new ValidationResult(true);
+        }
+
+        private boolean areForceMergeThreadsAvailable() {
+            for (ThreadPoolStats.Stats stats : threadPool.stats()) {
+                if (stats.getName().equals(ThreadPool.Names.FORCE_MERGE)) {
+                    this.maxConcurrentForceMerges = Math.max(1, stats.getThreads()) * forceMergeManagerSettings.getConcurrencyMultiplier();
+                    if ((stats.getQueue() == 0))
+                        // If force merge thread count is set by the customer( greater than 0) and active thread count is already equal or more than this threshold block skip any more force merges
+                        return forceMergeManagerSettings.getForceMergeThreadCount() <= 0 || stats.getActive() < forceMergeManagerSettings.getForceMergeThreadCount();
```

Comment:
`forceMergeManagerSettings.getForceMergeThreadCount() <= 0` Didn't get the reason for this check. if the value is 0 why are we returning true.